### PR TITLE
[NEW] Enhanced Mobile UI

### PR DESCRIPTION
**Pull request by billtt**
@RocketChat/core 

This PR improves the user experience on mobile devices:
* Use full screen for room list instead of half screen
* Enlarge room list item size and gap
* Close side nav when tapping on listed room
* Show room list by default on mobile devices (instead of home)
* Remove footer on room list
* Disable hover effects for room list and messages (on mobile devices, of course)

*original*

*modified*

### CODE REVIEW

File: packages/rocketchat-ui-master/client/main.js [hunk 1]
```diff
@@ -189,31 +189,32 @@ Template.main.events({
 			const diffY = touch.clientY - t.touchstartY;
 			const absX = Math.abs(diffX);
 			const absY = Math.abs(diffY);
+			const width = document.body.clientWidth;
```
Comment (karlprieb) [inline packages/rocketchat-ui-master/client/main.js]: You can use `window.matchMedia` here, it is more precise.

File: packages/rocketchat-ui/client/views/app/home.js [hunk 2]
```diff
@@ -6,3 +6,12 @@ Template.home.helpers({
 		return RocketChat.settings.get('Layout_Home_Body');
 	}
 });
+
+Template.home.onRendered(function() {
+	const width = document.body.clientWidth;
```
Comment (karlprieb) [inline packages/rocketchat-ui/client/views/app/home.js]: Use `window.matchMedia`

File: packages/rocketchat-theme/server/colors.less [hunk 3]
```diff
@@ -155,8 +155,14 @@
 	background-color: @transparent-darker;
 }
 
-.background-transparent-darker-hover:hover {
-	background-color: @transparent-darker;
+@media all and(min-width: 781px) {
+	.background-transparent-darker-hover:hover {
```
Comment (karlprieb) [inline packages/rocketchat-theme/server/colors.less]: You can't limit this global classes to desktop only, you need remove or add this class with JS when you target specific screen sizes.

**Comment by karlprieb** [comment]
@billtt Thank you for you contribution :)
I made some comments about RC best practices. Contact me if you need some help :)

File: packages/rocketchat-theme/server/colors.less (hunk 3 above)
Comment (billtt) [inline packages/rocketchat-theme/server/colors.less]: Wouldn't that be less efficient? I mean, running JS every time the page loads?

Comment (billtt) [inline packages/rocketchat-theme/server/colors.less]: And to explain more: I'm not limiting the global classes `background-transparent-darker-hover` and `background-transparent-dark-hover`, just limiting their hover effects to desktop .

File: packages/rocketchat-ui-master/client/main.js (hunk 1 above)
Comment (billtt) [inline packages/rocketchat-ui-master/client/main.js]: fixed :)

File: packages/rocketchat-ui/client/views/app/home.js (hunk 2 above)
Comment (billtt) [inline packages/rocketchat-ui/client/views/app/home.js]: fixed :)

**Comment by sampaiodiego** [comment]
I don't think we should trigger this new UI at the same `780px` as our media query does now.. 

also there is no way to slide the panel back, you need to find the channel you are so you can click on it. as of now we have the burger you can click to hide the left panel.

**Comment by rodrigok** [comment]
@billtt Can you fix the conflicts?

@karlprieb Can you review the changes against the new design?

**Comment by rodrigok** [comment]
We have a new design implemented now, this PR is no longer valid since some, or even all, improvements was already made.

Thanks by your contribution @billtt , we took this PR as base to our redesign.

**Pull request closed without merging**